Lib/test/test_pathlib.py

```diff
@@ -299,6 +293,11 @@ def test_repr_common(self):
             self.assertTrue(r.endswith(')'), r)
             inner = r[len(clsname) + 1 : -1]
             self.assertEqual(eval(inner), p.as_posix())
+
+    def test_repr_roundtrips(self):
+        for pathstr in ('a', 'a/b', 'a/b/c', '/', '/a/b', '/a/b/c'):
+            p = self.cls(pathstr)
+            r = repr(p)
             # The repr() roundtrips.
             q = eval(r, pathlib.__dict__)
             self.assertIs(q.__class__, p.__class__)
```

Comment:
Can you put all of this in a `self.subTest()` block?

https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests